specutils/spectra/spectral_axis.py

```diff
@@ -0,0 +1,98 @@
+        The coordinate (position and velocity) of observer.
+    target : `~astropy.coordinates.BaseCoordinateFrame` or `~astropy.coordinates.SkyCoord`, optional
+        The coordinate (position and velocity) of observer.
+    radial_velocity : `~astropy.units.Quantity`, optional
+        The radial velocity of the target with respect to the observer.
+    redshift : float, optional
+        The redshift of the target with respect to the observer.
+    doppler_rest : `~astropy.units.Quantity`, optional
+        The rest value to use for velocity space transformations.
+    doppler_convention : str, optional
+        The convention to use when converting the spectral data to/from
+        velocity space.
+    bin_specificatoin: str, optional
+        Must be "edges" or "centers". Determines whether specified axis values
+        are interpreted as bin edges or bin centers. Defaults to "centers".
+    """
+
+    def __new__(cls, value, unit=None, observer=None, target=None,
+                 radial_velocity=None, redshift=None, doppler_rest=None,
+                 doppler_convention=None, bin_specification="centers",
+                 **kwargs):
+
+        # Convert to bin centers if bin edges were given, since SpectralCoord
+        # only accepts centers
+        if bin_specification == "edges":
+            bin_edges = value
+            value = SpectralAxis._centers_from_edges(value)
+
+        obj = super().__new__(cls, value, unit=unit, observer=observer,
+                              target=target, radial_velocity=radial_velocity,
+                              redshift=redshift, doppler_rest=doppler_rest,
+                              doppler_convention=doppler_convention, **kwargs)
+
+        if bin_specification == "edges":
+            obj.bin_edges = bin_edges
+
+        return obj
+
+    def __quantity_subclass__(self, unit):
+        """
+        Overridden by subclasses to change what kind of view is
+        created based on the output unit of an operation.
+        """
+        return SpectralAxis, True
+
+    @staticmethod
+    def _edges_from_centers(centers):
+        """
+        Calculates interior bin edges based on the average of each pair of
+        centers, with the two outer edges based on extrapolated centers added
+        to the beginning and end of the spectral axis.
+        """
+        a = np.insert(centers, 0, 2*centers[0] - centers[1])
+        b = np.append(centers, 2*centers[-1] - centers[-2])
+        edges = (a + b) / 2
+        return edges
+
+    @staticmethod
+    def _centers_from_edges(edges):
+        """
+        Calculates the bin centers as the average of each pair of edges
+        """
+        return (edges[1:] + edges[:-1]) / 2
+
+    @lazyproperty
+    def bin_edges(self):
+        """
+        Calculates bin edges if the spectral axis was created with centers
+        specified.
+        """
+        return self._edges_from_centers(self.value)
```

Comment:
A minor quibble. I realize that the user should be aware of what specification their axis is in, however, if they initialized the `SpectralAxis` with `bin_specification='edges'` (as is default), then this gives wrong answer. I think we may want to check the state of the bin specification and return just `self.value` it's already `edges`.

```suggestion
        if self._bin_edges == 'edges':
            return self.value
            
        return self._edges_from_centers(self.value)
```

Reply:
Two things: 
- "centers" is default, not "edges"
- I decided to implement this such that no matter what is input, the bin centers are always the thing that are stored in self.value, and self.bin_edges will have either the specified bin edges (if they were provided) or the calculated bin edges (if centers were provided). I figured this would mesh better with all the current code that expects bin centers from a `spec.spectral_axis` while still making the bin edges available when desired, rather than having to constantly check which is being returned in `spec.spectral_axis`. I'm very open to discussion on whether this is the right strategy.

Reply:
Also, I wasn't sure if I grasped 100% what `@lazyproperty` does, but my hope was that if obj.bin_edges was already set then it would never be evaluated (and thus, here, only be evaluated if centers were specified).